src/main/src/main/java/org/geoserver/catalog/impl/CatalogImpl.java

```diff
@@ -1295,12 +1295,27 @@ public StyleInfo getStyleByName(String name) {
                 String name2 = name.replaceAll(REPLACE, "_");
                 // check workspace: sty_le
                 result = getStyleByName(prefix, resource2);
+                if (result != null) {
+                    return result;
+                }
                 // check workspace_sty_le
                 result = facade.getStyleByName(name2);
+                if (result != null) {
+                    return result;
+                }
+                LOGGER.log(Level.WARNING, "Style name " + name
+                        + " contains an invalid character. You may not see correct results. "
+                        + "Please update your catalog.");
```

Comment:
I think I would only log this message before each "return result" method below. Like only if we find something with an invalid name can we be sure the style contains an invalid name. It could be that the style does not exist at all ...